### PR TITLE
[`bnb`] add flan-t5 example

### PR DESCRIPTION
**Pull request by younesbelkada**
# What does this PR do?

This PR adds an example that leverages `peft` & `bitsandbytes` to fine-tune `flan-t5-large` on a financial dataset for sentiment classification

cc @pacman100 @sayakpaul 

Do not merge before https://github.com/huggingface/peft/pull/85

Related: https://github.com/huggingface/peft/issues/80

LoRA weights stored in: https://huggingface.co/ybelkada/flan-t5-large-financial-phrasebank-lora

### CODE REVIEW

**Comment by sayakpaul** [comment]
@younesbelkada thanks! Just let me know when this is good for a review.

**Comment by younesbelkada** [comment]
This is good for a review! 🔥 
After merging #85 I just need to quickly update the install requirements section

**Review by sayakpaul — COMMENTED** [review]
Thanks for the notebook 🔥 

Here are some comments:

* Maybe add an introduction on what task we'll be focusing on? You can take an example from the LoRA image classification notebook. 
* In the "Prepare model for training" section, add a few more details. I think the readers will appreciate it. 
* Spend a sentence emphasizing that we're only training ~0.6% of the total model params. 
* Not sure what `sentences_allagree` means in the `load_dataset()` method. Maybe spit out the argument name? 
* `labels[labels==tokenizer.pad_token_id] = -100` -- add a comment on why it's needed?
* Under model evaluation, do we want to obtain accuracy on the validation set? I think it's an easier metric to interpret. 
* No strong opinions here, but I find it aesthetic to run a linter extension (like [this one](https://github.com/drillan/jupyter-black)) on the notebook for formatting the code.

File: examples/int8_training/Finetune_flan_t5_large_bnb_peft.ipynb [hunk 1]
```diff
@@ -0,0 +1,8321 @@
+{
```
Comment (sayakpaul) [inline examples/int8_training/Finetune_flan_t5_large_bnb_peft.ipynb:1]: <pre class="ql-syntax" spellcheck="false">Line #5.    def preprocess_function(examples):
</pre><p>A space or two before the function would be nice. </p>
<br/>

 _Reply via <a href='https://app.reviewnb.com/huggingface/peft/pull/86/discussion/'>ReviewNB</a>_ <div id='ReviewNBCommentContext-DoNotDelete' style='display:none' data-state='OPEN' data-cellIndex='15'/>

**Comment by sayakpaul** [comment]
Thanks @younesbelkada! The changes look good to me. Just a minor comment but otherwise good to go from my side.

**Review by pacman100 — APPROVED** [review]
Thank you for adding this example, FlanT5 being best open source model, this example will greatly help the wider community! 🚀

LGTM!  🤗
